sync-core/src/main/java/com/cloudant/sync/replication/Filter.java

```diff
@@ -0,0 +1,140 @@
+ * when a pull replication calls the source database's {@code _changes}
+ * feed.</p>
+ *
+ * <p>For a filter function that takes no parameters, use this
+ * constructor:</p>
+ *
+ * <pre>
+ * Filter filter = new Filter("filterDoc/filterFunctionName")
+ * </pre>
+ *
+ * <p>For a filter function that requires parameters, use this
+ * constructor:</p>
+ *
+ * <pre>
+ * Map&lt;String, String&gt; params = new HashMap();
+ * map.put("max_age", "10");
+ * map.put("name", "john");
+ * Filter filter = new Filter("doc/filterName", map);
+ * </pre>
+ *
+ * @see <a href="http://couchdb.readthedocs.org/en/1.6.x/replication/intro.html#controlling-which-documents-to-replicate">Controlling documents replicated</a>
+ * @see <a href="http://docs.couchdb.org/en/1.6.x/couchapp/ddocs.html#filter-functions">Filter functions CouchDB docs</a>
+ */
+public class Filter {
+
+    private final String name;
+
+    private final Map<String, String> parameters;
+
+    /**
+     * Constructs a filter object for a function that requires no
+     * parameters.
+     *
+     *
+     *
+     *
+     * @param filterName filter function name. The {@code filterName} argument is the name of the
+     *                   filter, as passed to
+     *                   the {@code filter} parameter of CouchDB's {@code _changes} feed. This
+     *                   and the name of the filter function, separated by a slash. For example,
+     *                   {@code filterDoc/filterFunctionName}
+     */
+    public Filter(String filterName) {
+        this.name = filterName;
+        this.parameters = null;
+    }
+
+    /**
+     * Constructs a filter object for a function that requires no
+     * parameters.
+     *
+     *
+     *
+     *
+     * @param filterName filter function name. The {@code filterName} argument is the name of the
+     *                   filter, as passed to
+     *                   the {@code filter} parameter of CouchDB's {@code _changes} feed. This
+     *                   and the name of the filter function, separated by a slash. For example,
+     *                   {@code filterDoc/filterFunctionName}
+     *
+     * @param parameters Any parameters required for the function. Can be {@code null}. The contents
+     *                   of {@code properties} are expanded to {@code key=value} pairs when
+     *                   constructing the {@code _changes} feed call for the remote database.
+     *                   Integer values should be added as String objects.
+     *
+     * @see <a href="http://docs.couchdb.org/en/1.6.x/couchapp/ddocs.html#filter-functions">Filter functions CouchDB docs</a>
+     */
+    public Filter(String filterName, Map<String, String> parameters) {
+        this.name = filterName;
+        ImmutableMap.Builder<String, String> builder = new ImmutableMap.Builder<String, String>();
+        builder.putAll(parameters);
+        this.parameters = builder.build();
```

Comment:
We can do this without using Guava (presuming it's fine on Android):

```
this.parameters = Collections.unmodifiableMap(new Hashmap<>(parameters));
```
http://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#unmodifiableMap%28java.util.Map%29